src/main/java/webserver/request/RequestHeader.java

```diff
@@ -0,0 +1,86 @@
+package webserver.request;
+
+import webserver.HttpHeaders;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public final class RequestHeader {
+
+    private static final String DELIMITER = ":";
+    private static final String DEFAULT_CONTENT_LENGTH = "0";
+
+    private final Map<String, String> headers;
+
+    private RequestHeader(Map<String, String> headers) {
+        this.headers = Collections.unmodifiableMap(headers);
+    }
+
+    public static RequestHeader from(Map<String, String> headers) {
+        return new RequestHeader(Objects.requireNonNullElse(headers, Collections.emptyMap()));
+    }
+
+    public static RequestHeader from(List<String> headerStrings) {
+        if (headerStrings == null) {
+            return empty();
+        }
+        return from(
+                headerStrings.stream()
+                        .filter(string -> string.contains(DELIMITER))
+                        .map(RequestHeader::parseToEntry)
+                        .collect(Collectors.toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue))
+        );
+    }
+
+    private static RequestHeader empty() {
+        return from(Collections.emptyMap());
+    }
+
+    private static Map.Entry<String, String> parseToEntry(String keyValue) {
+        String[] splitKeyValue = keyValue.split(DELIMITER, 2);
+        return Map.entry(splitKeyValue[0].trim(), splitKeyValue[1].trim());
+    }
+
+    public Optional<String> value(String key) {
+        if (headers.containsKey(key)) {
+            return Optional.ofNullable(headers.get(key));
+        }
+        return Optional.empty();
+    }
```

Comment:
```suggestion
    public Optional<String> value(String key) {
        return Optional.ofNullable(headers.get(key));
    }
```

혹시 containsKey 를 한번 더 체크를 해주어야 하는 이유가 있는지, 코드를 읽다가 궁금해서 여쭈어봅니다.
ofNullable 에서 만약 null 일 경우 이미 Optional.empty() 를 리턴하는 것으로 알고 있어서요.
워낙 코드가 좋으신지라, 혹시 제가 Optional 에 관해서 모르는게 있는것 같아서 한수 배우고자 여쭈어봅니다.

Reply:
이 부분은 저도 고민이 되던 부분이었습니다. 🤔
코드를 보시다보면 `ofNullable` 사용한 경우도 있긴 했었는데요
기본적으로 `Map` 에서 `null` 을 반환할거라 예상 하지만 실제 구현 클래스 내부에서는 어떤 예외가 발생될지 예상이 안된다고 생각했습니다.

이처럼 [엘레강트 오브젝트](https://github.com/hyejineee/Reading-and-Writing/blob/master/%EC%A0%95%EB%A6%AC/%EC%97%98%EB%A0%88%EA%B0%95%ED%8A%B8%20%EC%98%A4%EB%B8%8C%EC%A0%9D%ED%8A%B8.md#42-%EC%B2%B4%ED%81%AC-%EC%98%88%EC%99%B8%EB%A7%8C-%EB%8D%98%EC%A7%80%EC%84%B8%EC%9A%94) 에서는 예외를 명시적으로 알기 위해서는 체크 예외만 사용하라고 권장하지만 현실적으로 그러지 못한 경우가 많으며,  

[이펙티브 자바](https://madplay.github.io/post/effectivejava-chapter10-exceptions#%EC%95%84%EC%9D%B4%ED%85%9C-69-%EC%98%88%EC%99%B8%EB%8A%94-%EC%A7%84%EC%A7%9C-%EC%98%88%EC%99%B8-%EC%83%81%ED%99%A9%EC%97%90%EB%A7%8C-%EC%82%AC%EC%9A%A9%ED%95%98%EB%9D%BC)에서는 제어 흐름을 위해서는 예외보다 상태 검사 메서드 제공을 권장하고 있었습니다.

여기서 `null` 이 반환되는 경우도 nullSafety 환경에서 예외가 발생된 것이라고 생각한다면 
상태 검사 메서드(`containsKey`)를 이용하여 명시적으로 표현하는 것이 올바르지 않았을까 생각했습니다!! 
또한 그럼에도 조건절 안에서도 `ofNullable`을 사용한 이유는 코드를 한층 더 보호하기 위함이었습니다!